src/js/relayclient/RelayClient.js

```diff
@@ -441,23 +459,20 @@ class RelayClient {
           approvalData.length +
           relayCallExtraBytes
 
-        const _gasLimits = await gasSponsorContract.methods.getGasLimitsForSponsorCalls().call()
-        const _overhead = parseInt(await relayHub.methods.getHubOverhead().call())
-        const gasData = await utils.getTransactionGasData({
-          _gasLimits,
-          _overhead,
-          relayHub,
-          calldataSize,
-          gtxdatanonzero: options.gtxdatanonzero || Environments.default.gtxdatanonzero,
+        const gasLimits = await paymasterContract.methods.getGasLimits().call()
```

Comment:
this is far too much work to do before `canRelay`, just like you don't do all these calculations before calling `relayCall`, you shouldn't here:
need to have a wrapper for canRelay which get all the needed vallues (hub overhead, gasLimits etc))

Reply:
Will do. I think this will need to be done in the scope of StakeManager separation, so didn't make the change here.